app/models/katello/glue/candlepin/pool.rb

```diff
@@ -6,179 +6,128 @@ def self.included(base)
       base.send :extend, ClassMethods
 
       base.class_eval do
-        lazy_accessor :remote_data, :pool_derived, :product_name, :consumed, :quantity, :available, :support_level, :support_type,
-          :start_date, :end_date, :attrs, :owner, :product_id, :account_number, :contract_number,
-          :source_pool_id, :host_id, :virt_only, :virt_limit, :multi_entitlement, :stacking_id,
-          :arch, :sockets, :cores, :ram, :description, :product_family, :variant, :provided_products,
-          :active, :instance_multiplier, :suggested_quantity, :unmapped_guest,
-          :initializer => (lambda do |_s|
-                             json = Resources::Candlepin::Pool.find(cp_id)
-                             # symbol "attributes" is reserved by Rails and cannot be used
-                             json['attrs'] = json['attributes']
-                             json
-                           end)
+        lazy_accessor :pool_derived, :owner, :source_pool_id, :host_id, :virt_only, :virt_limit, :arch, :description,
+          :product_family, :variant, :suggested_quantity, :unmapped_guest, :support_type, :product_id, :type,
+          :initializer => lambda { |_s| self.import_lazy_attributes }
+
+        lazy_accessor :active, :initializer => lambda { |_s| self.backend_data["activeSubscription"] }
+
+        lazy_accessor :available, :initializer => lambda { |_s| self.quantity_available }
+
+        lazy_accessor :name, :support_level, :org, :sockets, :cores, :stacking_id, :instance_multiplier,
+          :initializer => lambda { |_s| self.subscription ? self.subscription.attributes : {} }
       end
     end
 
     module ClassMethods
-      def find_by_organization_and_id(organization, pool_id)
-        pool = Katello::Pool.find_by_cp_id(pool_id.to_s) || Pool.new(Resources::Candlepin::Pool.find(pool_id))
-        if pool.organization == organization
-          return pool
-        end
+      def candlepin_data(cp_id)
+        Katello::Resources::Candlepin::Pool.find(cp_id)
       end
 
-      def find_by_organization_and_id!(organization, pool_id)
-        subscription = find_by_organization_and_id(organization, pool_id)
-        fail ActiveRecord::RecordNotFound if subscription.nil?
-        subscription
+      def get_for_owner(organization)
+        Katello::Resources::Candlepin::Pool.get_for_owner(organization)
       end
 
-      def find_by_id(pool_id)
-        Katello::Pool.find_by_cp_id(pool_id) || Pool.new(Resources::Candlepin::Pool.find(pool_id))
-      end
-
-      def find_by_id!(pool_id)
-        subscription = find_by_id(pool_id)
-        fail ActiveRecord::RecordNotFound if subscription.nil?
-        subscription
+      def index_pool(cp_pool_id)
```

Comment:
Is this function still needed? Could it be renamed to follow the import convention?